aospy/test/test_utils_times.py

```diff
@@ -485,5 +486,51 @@ def test_sel_time(self):
         self.assertEqual(result[SUBSET_END_DATE_STR].values, end_date)
 
 
+def test_yearly_average_no_mask():
+    times = pd.to_datetime(['2000-06-01', '2000-06-15',
+                            '2001-07-04', '2001-10-01', '2001-12-31',
+                            '2004-01-01'])
+    arr = xr.DataArray(np.random.random((len(times),)),
+                       dims=[TIME_STR], coords={TIME_STR: times})
+    dt = arr.copy(deep=True)
+    dt.values = np.random.random((len(times),))
+
+    actual = yearly_average(arr, dt)
+
+    yr2000 = (arr[0]*dt[0] + arr[1]*dt[1]) / (dt[0] + dt[1])
+    yr2001 = ((arr[2]*dt[2] + arr[3]*dt[3] + arr[4]*dt[4]) /
+              (dt[2] + dt[3] + dt[4]))
+    yr2004 = arr[-1]
+    yrs_coord = [2000, 2001, 2004]
+    yr_avgs = np.array([yr2000, yr2001, yr2004])
+    desired = xr.DataArray(yr_avgs, dims=['year'], coords={'year': yrs_coord})
+    assert np.allclose(actual.values, desired.values)
```

Comment:
Both here and in the masked test, could we use `xr.testing.assert_identical(actual, desired)`?  I think this would check the array values and year coordinate all in one line.

Reply:
Good call; forgot about `xr.testing`.  I'll use `xr.testing.assert_allclose` though; in some cases I was getting failures because my by-hand averaging weren't bitwise identical with the xarray/numpy methods.